web/modules/custom/exchange_rates/src/Plugin/Block/ExchangeRatesBlock.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Drupal\exchange_rates\Plugin\Block;
+
+use Drupal\Core\Block\BlockBase;
+
+/**
+ * Provides data about currency exchange rates.
+ *
+ * @Block(
+ *   id = "exchange_rates",
+ *   admin_label = @Translation("Exchange_rates"),
+ * )
+ */
+class ExchangeRatesBlock extends BlockBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    $httpClient = \Drupal::httpClient();
+    $date = date("Ymd");
+    $exchangeRates =
+      json_decode($httpClient
+        ->get('https://bank.gov.ua/NBUStatService/v1/statdirectory/exchange?date=' . $date . '&json')
+        ->getBody());
```

Comment:
1) get the body of a response
2) then decode it

Reply:
resolved